packages/twenty-server/src/workspace/calendar/services/google-calendar-full-sync.service.ts

```diff
@@ -0,0 +1,270 @@
+
+    const refreshToken = connectedAccount.refreshToken;
+    const workspaceMemberId = connectedAccount.accountOwnerId;
+
+    if (!refreshToken) {
+      throw new Error(
+        `No refresh token found for connected account ${connectedAccountId} in workspace ${workspaceId} during full-sync`,
+      );
+    }
+
+    const calendarChannel =
+      await this.calendarChannelService.getFirstByConnectedAccountIdOrFail(
+        connectedAccountId,
+        workspaceId,
+      );
+
+    const calendarChannelId = calendarChannel.id;
+
+    const googleCalendarClient =
+      await this.googleCalendarClientProvider.getGoogleCalendarClient(
+        refreshToken,
+      );
+
+    const isBlocklistEnabledFeatureFlag =
+      await this.featureFlagRepository.findOneBy({
+        workspaceId,
+        key: FeatureFlagKeys.IsBlocklistEnabled,
+        value: true,
+      });
+
+    const isBlocklistEnabled =
+      isBlocklistEnabledFeatureFlag && isBlocklistEnabledFeatureFlag.value;
+
+    const blocklist = isBlocklistEnabled
+      ? await this.blocklistService.getByWorkspaceMemberId(
+          workspaceMemberId,
+          workspaceId,
+        )
+      : [];
+
+    const blocklistedEmails = blocklist.map((blocklist) => blocklist.handle);
+    let startTime = Date.now();
+
+    const googleCalendarEvents = await googleCalendarClient.events.list({
+      calendarId: 'primary',
+      maxResults: 500,
+      pageToken: pageToken,
+      q: googleCalendarSearchFilterExcludeEmails(blocklistedEmails),
+    });
+
+    let endTime = Date.now();
+
+    this.logger.log(
+      `google calendar full-sync for workspace ${workspaceId} and account ${connectedAccountId} getting events list in ${
+        endTime - startTime
+      }ms.`,
+    );
+
+    const {
+      items: events,
+      nextPageToken,
+      nextSyncToken,
+    } = googleCalendarEvents.data;
+
+    if (!events || events?.length === 0) {
+      this.logger.log(
+        `google calendar full-sync for workspace ${workspaceId} and account ${connectedAccountId} done with nothing to import.`,
+      );
+
+      return;
+    }
+
+    const eventExternalIds = events.map((event) => event.id as string);
+
+    startTime = Date.now();
+
+    const existingCalendarChannelEventAssociations =
+      await this.calendarChannelEventAssociationService.getByEventExternalIdsAndCalendarChannelId(
+        eventExternalIds,
+        calendarChannelId,
+        workspaceId,
+      );
+
+    endTime = Date.now();
+
+    this.logger.log(
+      `google calendar full-sync for workspace ${workspaceId} and account ${connectedAccountId}: getting existing calendar channel event associations in ${
+        endTime - startTime
+      }ms.`,
+    );
+
+    // TODO: In V2, we will also import deleted events by doing batch GET queries on the canceled events
+    // The canceled events start and end are not accessible in the list query
+    const formattedEvents = events
+      .filter((event) => event.status !== 'cancelled')
```

Comment:
Just realised that I think we use "canceled" in the code, right? Is it handled properly?

Reply:
It is :) https://github.com/twentyhq/twenty/pull/4442/files#diff-65fa7d566d378d5546e391a1e942301dac777ae21f51cb89f63aec96605c3b52R15